src/sparse/impl/KokkosSparse_twostage_gauss_seidel_impl.hpp

```diff
@@ -0,0 +1,919 @@
+            if (column_view (k) < i) {
+              nnzL ++;
+            } else if (column_view (k) == i && !two_stage) {
+              nnzL ++;
+              nnzU ++;
+            } else if (column_view (k) > i && column_view (k) < num_rows) {
+              nnzU ++;
+            }
+          }
+          row_map (i+1) = nnzL;
+          row_map2 (i+1) = nnzU;
+          if (i == 0) {
+            row_map (0) = 0;
+            row_map2 (0) = 0;
+          }
+          nnz += nnzL;
+        }
+
+        // functor for storing entriesL and entriesU (with parallel_for)
+        KOKKOS_INLINE_FUNCTION
+        void operator()(const Tag_entriesLU&, const ordinal_t i) const
+        {
+          ordinal_t nnzL = row_map (i);
+          ordinal_t nnzU = row_map2 (i);
+          for (size_type k = rowmap_view (i); k < rowmap_view (i+1); k++) {
+            if (column_view (k) < i) {
+              entries (nnzL) = column_view (k);
+              nnzL ++;
+            } else if (column_view (k) == i && !two_stage) {
+              entries (nnzL) = column_view (k);
+              entries2 (nnzU) = column_view (k);
+              nnzL ++;
+              nnzU ++;
+            } else if (column_view (k) > i && column_view (k) < num_rows) {
+              entries2 (nnzU) = column_view (k);
+              nnzU ++;
+            }
+          }
+        }
+
+        // functor for storing both valuesL & valuesU (with parallel_for)
+        KOKKOS_INLINE_FUNCTION
+        void operator()(const Tag_valuesLU&, const ordinal_t i) const
+        {
+          const_scalar_t one (1.0);
+          ordinal_t nnzL = row_map (i);
+          ordinal_t nnzU = row_map2 (i);
+          for (size_type k = rowmap_view (i); k < rowmap_view (i+1); k++) {
+            if (column_view (k) < i) {
+              values (nnzL) = values_view (k);
+              nnzL ++;
+            } else if (column_view (k) == i) {
+              if (two_stage) {
+                diags (i) = one / values_view (k);
+              } else {
+                values (nnzL) = values_view (k);
+                values2 (nnzU) = values_view (k);
+                nnzL ++;
+                nnzU ++;
+              }
+            } else if (column_view (k) < num_rows) {
+              values2 (nnzU) = values_view (k);
+              nnzU ++;
+            }
+          }
+          #if defined(KOKKOSSPARSE_IMPL_TWOSTAGE_GS_MERGE_SPMV)
+          if (two_stage) {
+            for (ordinal_t k = row_map (i); k < nnzL; k++) {
+              values (k) *= diags (i);
+            }
+            for (ordinal_t k = row_map2 (i); k < nnzU; k++) {
+              values2 (k) *= diags (i);
+            }
+          }
+          #endif
+        }
+
+
+        // ------------------------------------------------------- //
+        // functor for computinr row_map (with parallel_scan)
+        KOKKOS_INLINE_FUNCTION
+        void operator() (const Tag_compPtr&, const ordinal_t i, ordinal_t &update, const bool final) const
```

Comment:
Could this be replaced with ``KokkosKernels::Impl::kk_inclusive_parallel_prefix_sum`` in common/KokkosKernels_SimpleUtils.hpp?